rcl/include/rcl/publisher.h

```diff
@@ -336,6 +340,29 @@ RCL_WARN_UNUSED
 rmw_publisher_t *
 rcl_publisher_get_rmw_handle(const rcl_publisher_t * publisher);
 
+/// Check that the publisher is valid
+/**
+ * The bool returned is `false` if:
+ *   - the argument `publisher` is `NULL`
+ *   - the argument's `publisher->impl` is `NULL`
```

Comment:
The list of reasons for client is different (more exhaustive). I don't know if we need the list to be that exhaustive, but it ought to be consistent amongst similar elements.